rollup-node.md

```diff
@@ -0,0 +1,285 @@
+[deposits]: /glossary.md#deposits
+[deposit-feed]: /glossary.md#L2-deposit-feed-contract
+[L2 chain inception]: /glossary.md#L2-chain-inception
+
+The [rollup node] is the component responsible for [deriving the L2
+chain][derivation] from L1 blocks. This process happens in two steps:
+
+1. Read [L2 derivation inputs] from L1 and generate [payload attributes] (essentially
+   [a block without output properties][block]).
+2. Pass the payload attributes to the [execution engine], so that [output block
+   properties][block] may be computed.
+
+While this process is conceptually a pure function from the L1 chain to the L2
+chain, it is in practice incremental. The L2 chain changes whenever new L1
+blocks (and hence new L2 derivation inputs) are added to the L1 chain, or
+whenever the L1 chain [re-organizes][reorg].
+
+Additionally, the rollup node is responsible for [block gossip], i.e.
+transmitting the L2 blocks it derives over a peer-to-peer network.
+
+The part of the rollup node that derives the L2 chain is called the [rollup
+driver]. This document is mostly concerned with the specification of the rollup
+driver, while block gossip is [specified in a separate document][gossip-spec].
+
+[gossip-spec]: TODO
+
+> **TODO LINK** block gossip spec
+
+## L2 Chain Derivation
+[l2-chain-derivation]: #l2-chain-derivation
+
+This section specifies how the [rollup driver] derives one L2 block from every
+L1 block. The L2 block will carry the L1 block attributes (as a *[L1 attributes
+transaction]*) well as all L2 transactions deposited by users in the L1 block
+(*[deposits]*).
+
+[L1 attributes transaction]: /glossary.md#l1-attributes-transaction
+
+### From L2 derivation inputs to payload attributes
+
+[payload-attr]: #From-L2-derivation-inputs-to-payload-attributes
+[`PayloadAttributesOPV1`]: #From-L2-derivation-inputs-to-payload-attributes
+
+The rollup reads the following [L2 derivation inputs] for each L1 block:
+
+- L1 block attributes
+   - block number
+   - timestamp
+   - basefee
+- [deposits]
+
+A deposit is an L2 transaction that has been submitted on L1, via a transaction
+sent to the [L2 deposit feed contract][deposit-feed].
+
+While deposits are notably (but not only) used to "deposit" (bridge) ETH and
+tokens to L2, the word *deposit* should be understood as "a transaction
+*deposited* to L2".
+
+The L1 attributes are read from the L1 block header, while deposits are read
+from the block's log entries. Refer to the [**L2 deposit feed contract
+specification**][deposit-feed] for details on how deposits are encoded as
+log entries.
+
+[deposit-feed-spec]: TODO
+
+> **TODO LINK** deposit feed contract specification
+
+From these L2 derivation inputs, the rollup node constructs an expanded version
+of the [Engine API PayloadAttributesV1 object][PayloadAttributesV1]:
+
+[PayloadAttributesV1]: https://github.com/ethereum/execution-apis/blob/main/src/engine/specification.md#payloadattributesv1
+
+```
+PayloadAttributesOPV1: {
```

Comment:
We should move this into its own block-derivation doc to keep this doc light enough to cover all rollup-node responsibilities.

Reply:
I don't get it, block derivation is 90% of the content of the present document, so you would move the whole document??